app/common/app_main.c

```diff
@@ -165,8 +168,10 @@ void app_init() {
 
 }
 
+#ifdef __cplusplus
```

Comment:
We should remove this block then. This part is never going to be compiled for cpp tests.